core/src/main/java/org/apache/accumulo/core/util/AccumuloUncaughtExceptionHandler.java

```diff
@@ -26,10 +26,20 @@
 public class AccumuloUncaughtExceptionHandler implements UncaughtExceptionHandler {
 
   private static final Logger log = LoggerFactory.getLogger(AccumuloUncaughtExceptionHandler.class);
+  private static final String HALT_PROPERTY = "HaltVMOnThreadError";
 
   @Override
   public void uncaughtException(Thread t, Throwable e) {
-    log.error(String.format("Caught an exception in %s.  Shutting down.", t), e);
+    if (e instanceof Exception) {
+      log.error(String.format("Caught an exception in %s. Thread is dead.", t), e);
+    } else {
+      if (System.getProperty(HALT_PROPERTY, "false").equals("true")) {
+        log.error(String.format("Caught an exception in %s.", t), e);
```

Comment:
We don't need String.format here. The slf4j logger supports substitution with `{}`

```suggestion
        log.error("Caught an exception in {}.", t, e);
```